node-manager/src/event.rs

```diff
@@ -0,0 +1,504 @@
+                    "event",
+                    "",
+                    0,
+                ));
+            }
+            Event::HTLCHandlingFailed { .. } => {
+                self.logger.log(&Record::new(
+                    lightning::util::logger::Level::Debug,
+                    format_args!("EVENT: HTLCHandlingFailed ignored"),
+                    "event",
+                    "",
+                    0,
+                ));
+            }
+            Event::PendingHTLCsForwardable { time_forwardable } => {
+                self.logger.log(&Record::new(
+                    lightning::util::logger::Level::Debug,
+                    format_args!("EVENT: PendingHTLCsForwardable processing"),
+                    "event",
+                    "",
+                    0,
+                ));
+                let forwarding_channel_manager = self.channel_manager.clone();
+                let min = time_forwardable.as_millis() as i32;
+                spawn_local(async move {
+                    sleep(min).await;
+                    forwarding_channel_manager.process_pending_htlc_forwards();
+                });
+            }
+            Event::SpendableOutputs { outputs } => {
+                self.logger.log(&Record::new(
+                    lightning::util::logger::Level::Debug,
+                    format_args!("EVENT: SpendableOutputs processing"),
+                    "event",
+                    "",
+                    0,
+                ));
+                let outputs_thread = outputs.clone();
+                let wallet_thread = self.wallet.clone();
+                let keys_manager_thread = self.keys_manager.clone();
+                let chain_thread = self.chain.clone();
+                spawn_local(async move {
+                    let destination_address = wallet_thread
+                        .wallet
+                        .lock()
+                        .await
+                        .get_address(AddressIndex::New)
+                        .unwrap();
+
+                    let output_descriptors = &outputs_thread.iter().collect::<Vec<_>>();
+                    let tx_feerate = fallback_fee_from_conf_target(ConfirmationTarget::Normal);
+                    let spending_tx = keys_manager_thread
+                        .spend_spendable_outputs(
+                            output_descriptors,
+                            Vec::new(),
+                            destination_address.script_pubkey(),
+                            tx_feerate,
+                            &Secp256k1::new(),
+                        )
+                        .unwrap();
+                    chain_thread.broadcast_transaction(&spending_tx);
+                });
+            }
+            Event::ChannelClosed {
+                channel_id,
+                reason,
+                user_channel_id: _,
+            } => {
+                self.logger.log(&Record::new(
```

Comment:
channel manager stuff? no we don't ever delete channel information ourselves. LDK should handle that. For instance, its not confirmed yet tho it did close. It may need that info.